src/libtriton/engines/solver/z3/z3Solver.cpp

```diff
@@ -123,18 +122,24 @@ namespace triton {
               /* Uniq result */
               if (exp.get_sort().is_bv())
                 args.push_back(ctx.bv_const(varName.c_str(), bvSize) != ctx.bv_val(svalue.c_str(), bvSize));
-
             }
 
-            /* Escape last models */
-            solver.add(triton::engines::solver::mk_or(args));
+            /* Check that model is available */
+            if (smodel.empty())
+              break;
+
+            /* Push model */
+            ret.push_back(smodel);
 
-            /* If there is model available */
-            if (smodel.size() > 0)
-              ret.push_back(smodel);
+            if (--limit) {
+              /* Escape last models */
+              if (!args.empty()) {
+                solver.add(triton::engines::solver::mk_or(args));
```

Comment:
As we will probably have Bitwuzla into Triton soon, can you move `::solver::mk_or` into a new `z3` namespace like `::solver::z3::mk_or` ?

Reply:
z3 already has its one `z3` namespace. I suppose this can mess up the namespaces:
```
/home/vishnya/fwork/sydr/triton/src/libtriton/engines/solver/z3/z3Solver.cpp:272:62: error: ‘check_result’ is not a member of ‘triton::engines::solver::z3’
       void Z3Solver::writeBackStatus(z3::solver& solver, z3::check_result res, triton::engines::solver::status_e* status) const {

```

Reply:
I can make it a member function of Z3Solver

Reply:
Yes, a static method then?

Reply:
Well, indeed. If it's only used in `Z3Solver` we can add it as private method.

Reply:
I pushed a fix.